grafana/grafana
# InfluxDB: test datasource

**What this PR does / why we need it**:
Part of https://github.com/grafana/grafana/issues/43076

This PR contains the changes needed to test the influxdb datasource while running queries through `api/ds/query`.

**Test scenarios**:

1. Run make devenv sources=influxdb,influxdb1,grafana grafana_version=8.2.4 (or other version before changes)
2. Open up http://localhost:3000 && http://127.0.0.1:3001 (to test older 8.2.4 version)
3. Select gdev-influxdb-influxql
   - Click the `Test` button on the config editor. The query should run through `api/ds/query`.
   - Go the the dashboard. Query should run through `api/ds/query`.
   - Go to edit mode. Query should run through `api/ds/query`.
   - Open up both supplied dashboards (Datasource tests - InfluxDB Templated && Datasource tests - InfluxDB Logs). Both should run queries through `api/ds/query`.
4. Select gdev-influxdb1-influxql
    - Should behave like `gdev-influxdb-influxql`

PS: make sure 
[feature_toggles]
enable = influxdbBackendMigration
